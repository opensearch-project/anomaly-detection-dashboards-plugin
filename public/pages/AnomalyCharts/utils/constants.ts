/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

export const INITIAL_ANOMALY_SUMMARY = {
  anomalyOccurrence: 0,
  minAnomalyGrade: 0.0,
  maxAnomalyGrade: 0.0,
  minConfidence: 0.0,
  maxConfidence: 0.0,
  lastAnomalyOccurrence: '',
};

export enum CHART_FIELDS {
  PLOT_TIME = 'plotTime',
  ANOMALY_GRADE = 'anomalyGrade',
  CONFIDENCE = 'confidence',
  DATA = 'data',
  AGG_INTERVAL = 'aggInterval',
  EXPECTED_VALUE = 'expectedValue',
}

export enum CHART_COLORS {
  ANOMALY_GRADE_COLOR = '#D13212',
  FEATURE_DATA_COLOR = '#16191F',
  FEATURE_COLOR = '#fcd529',
  CONFIDENCE_COLOR = '#017F75',
  LIGHT_BACKGROUND = '#FFFFFF',
  DARK_BACKGROUND = '#1D1E24',
}

export const FEATURE_CHART_THEME = [
  {
    lineSeriesStyle: {
      line: {
        strokeWidth: 2,
        visible: true,
        opacity: 0.5,
      },
      point: {
        visible: true,
        stroke: CHART_COLORS.FEATURE_DATA_COLOR,
      },
    },
  },
  {
    colors: {
      vizColors: [CHART_COLORS.FEATURE_DATA_COLOR],
    },
  },
];

export const ANOMALY_CHART_THEME = [
  {
    colors: {
      vizColors: [
        CHART_COLORS.CONFIDENCE_COLOR,
        CHART_COLORS.ANOMALY_GRADE_COLOR,
      ],
    },
  },
];

export const DATE_PICKER_QUICK_OPTIONS = [
  { start: 'now-24h', end: 'now', label: 'last 24 hours' },
  { start: 'now-7d', end: 'now', label: 'last 7 days' },
  { start: 'now-30d', end: 'now', label: 'last 30 days' },
  { start: 'now-90d', end: 'now', label: 'last 90 days' },

  { start: 'now/d', end: 'now', label: 'Today' },
  { start: 'now/w', end: 'now', label: 'Week to date' },
  { start: 'now/M', end: 'now', label: 'Month to date' },
  { start: 'now/y', end: 'now', label: 'Year to date' },
];

export enum LIVE_CHART_CONFIG {
  REFRESH_INTERVAL_IN_SECONDS = 30 * 1000, //poll anomaly result every 30 seconds
  MONITORING_INTERVALS = 60,
}

export const DEFAULT_ANOMALY_SUMMARY = {
  anomalyOccurrence: 0,
  minAnomalyGrade: 0,
  maxAnomalyGrade: 0,
  minConfidence: 0,
  maxConfidence: 0,
  lastAnomalyOccurrence: '-',
};

export const HEATMAP_CHART_Y_AXIS_WIDTH = 30;
