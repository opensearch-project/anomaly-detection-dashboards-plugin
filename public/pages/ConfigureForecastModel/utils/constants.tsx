/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import {
  ModelConfigurationFormikValues,
} from '../../ConfigureForecastModel/models/interfaces';
import { DEFAULT_SHINGLE_SIZE } from '../../../utils/constants';

// FIXME: We intentionally leave some parameters undefined to encourage customers to click 
// the "Suggest parameters" button rather than providing default values like "10 minutes" for interval
export const INITIAL_MODEL_CONFIGURATION_VALUES: ModelConfigurationFormikValues =
  {
    shingleSize: DEFAULT_SHINGLE_SIZE,
    imputationOption: undefined,
    interval: undefined,
    windowDelay: undefined,
    suggestedSeasonality: undefined,
    recencyEmphasis: undefined,
    resultIndex: undefined,
    resultIndexMinAge: 7,
    resultIndexMinSize: 51200,
    resultIndexTtl: 60,
    flattenCustomResultIndex: false,
  };

// an enum for the sparse data handling options
export enum SparseDataOptionValue {
  IGNORE = 'ignore',
  PREVIOUS_VALUE = 'previous_value',
  SET_TO_ZERO = 'set_to_zero',
  CUSTOM_VALUE = 'custom_value',
}
