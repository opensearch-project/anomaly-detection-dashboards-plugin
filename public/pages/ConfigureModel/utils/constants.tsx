/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import { FEATURE_TYPE } from '../../../models/interfaces';
import {
  ModelConfigurationFormikValues,
  FeaturesFormikValues,
} from '../../ConfigureModel/models/interfaces';
import { DEFAULT_SHINGLE_SIZE } from '../../../utils/constants';

export const INITIAL_MODEL_CONFIGURATION_VALUES: ModelConfigurationFormikValues =
  {
    featureList: [],
    categoryFieldEnabled: false,
    categoryField: [],
    shingleSize: DEFAULT_SHINGLE_SIZE,
    imputationOption: undefined
  };

export const INITIAL_FEATURE_VALUES: FeaturesFormikValues = {
  featureId: '',
  featureName: '',
  featureEnabled: true,
  featureType: FEATURE_TYPE.SIMPLE,
  aggregationQuery: JSON.stringify(
    { aggregation_name: { sum: { field: 'field_name' } } },
    null,
    4
  ),
  aggregationBy: '',
  aggregationOf: [],
};

export const FEATURE_TYPES = [
  { text: 'Custom Aggregation', value: FEATURE_TYPE.CUSTOM },
  { text: 'Defined Aggregation', value: FEATURE_TYPE.SIMPLE },
];

export const FEATURE_TYPE_OPTIONS = [
  { text: 'Field value', value: FEATURE_TYPE.SIMPLE },
  { text: 'Custom expression', value: FEATURE_TYPE.CUSTOM },
];

export enum SAVE_FEATURE_OPTIONS {
  START_AD_JOB = 'start_ad_job',
  KEEP_AD_JOB_STOPPED = 'keep_ad_job_stopped',
}

export const AGGREGATION_TYPES = [
  { value: 'avg', text: 'average()' },
  { value: 'value_count', text: 'count()' },
  { value: 'sum', text: 'sum()' },
  { value: 'min', text: 'min()' },
  { value: 'max', text: 'max()' },
];

export const FEATURE_FIELDS = [
  'featureName',
  'aggregationOf',
  'aggregationBy',
  'aggregationQuery',
];

// an enum for the sparse data handling options
export enum SparseDataOptionValue {
  IGNORE = 'ignore',
  PREVIOUS_VALUE = 'previous_value',
  SET_TO_ZERO = 'set_to_zero',
  CUSTOM_VALUE = 'custom_value',
}
