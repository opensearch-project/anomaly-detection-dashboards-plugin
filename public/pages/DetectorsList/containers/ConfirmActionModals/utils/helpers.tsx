/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';
import { EuiLink, EuiText, EuiIcon, EuiDataGrid } from '@elastic/eui';
import { getAlertingMonitorListLink } from '../../../../../utils/utils';
import { Monitor } from '../../../../../models/interfaces';
import { DetectorListItem } from '../../../../../models/interfaces';
import { PLUGIN_NAME } from '../../../../../utils/constants';
import { get, isEmpty } from 'lodash';
import { DETECTOR_STATE } from '../../../../../../server/utils/constants';

const getNames = (detectors: DetectorListItem[]) => {
  let data = [];
  for (let i = 0; i < detectors.length; i++) {
    data.push({
      Detector: (
        <EuiLink
          href={`${PLUGIN_NAME}#/detectors/${detectors[i].id}`}
          target="_blank"
        >
          {detectors[i].name} />
        </EuiLink>
      ),
    });
  }
  return data;
};

const getNamesAndMonitors = (
  detectors: DetectorListItem[],
  monitors: { [key: string]: Monitor }
) => {
  let data = [];
  for (let i = 0; i < detectors.length; i++) {
    const relatedMonitor = get(monitors, `${detectors[i].id}`);
    if (relatedMonitor) {
      data.push({
        Detector: (
          <EuiLink
            href={`${PLUGIN_NAME}#/detectors/${detectors[i].id}`}
            target="_blank"
          >
            {detectors[i].name} />
          </EuiLink>
        ),
        Monitor: (
          <EuiLink
            href={`${getAlertingMonitorListLink()}/${relatedMonitor.id}`}
            target="_blank"
          >
            {relatedMonitor.name} />
          </EuiLink>
        ),
      });
    } else {
      data.push({
        Detector: (
          <EuiLink
            href={`${PLUGIN_NAME}#/detectors/${detectors[i].id}`}
            target="_blank"
          >
            {detectors[i].name} />
          </EuiLink>
        ),
        Monitor: '-',
      });
    }
  }
  return data;
};

const getNamesAndMonitorsAndStates = (
  detectors: DetectorListItem[],
  monitors: { [key: string]: Monitor }
) => {
  let data = [];
  for (let i = 0; i < detectors.length; i++) {
    const relatedMonitor = get(monitors, `${detectors[i].id}`);
    const isRunning =
      detectors[i].curState === DETECTOR_STATE.INIT ||
      detectors[i].curState === DETECTOR_STATE.RUNNING;
    if (relatedMonitor) {
      data.push({
        Detector: (
          <EuiLink
            href={`${PLUGIN_NAME}#/detectors/${detectors[i].id}`}
            target="_blank"
          >
            {detectors[i].name} />
          </EuiLink>
        ),
        Monitor: (
          <EuiLink
            href={`${getAlertingMonitorListLink()}/${relatedMonitor.id}`}
            target="_blank"
          >
            {relatedMonitor.name} />
          </EuiLink>
        ),
        Running: <EuiText>{isRunning ? 'Yes' : 'No'}</EuiText>,
      });
    } else {
      data.push({
        Detector: (
          <EuiLink
            href={`${PLUGIN_NAME}#/detectors/${detectors[i].id}`}
            target="_blank"
          >
            {detectors[i].name}
          </EuiLink>
        ),
        Monitor: '-',
        Running: <EuiText>{isRunning ? 'Yes' : 'No'}</EuiText>,
      });
    }
  }
  return data;
};

export const getNamesGrid = (detectors: DetectorListItem[]) => {
  const gridData = getNames(detectors);
  return (
    <EuiDataGrid
      aria-label="Detector names"
      columns={[
        {
          id: 'Detector',
          isResizable: false,
          isExpandable: false,
          isSortable: false,
        },
      ]}
      columnVisibility={{
        visibleColumns: ['Detector'],
        setVisibleColumns: () => {},
      }}
      rowCount={gridData.length}
      renderCellValue={({ rowIndex, columnId }) =>
        //@ts-ignore
        gridData[rowIndex][columnId]
      }
      gridStyle={{
        border: 'horizontal',
        header: 'shade',
        rowHover: 'highlight',
        stripes: true,
      }}
      toolbarVisibility={false}
    />
  );
};

export const getNamesAndMonitorsGrid = (
  detectors: DetectorListItem[],
  monitors: { [key: string]: Monitor }
) => {
  const gridData = getNamesAndMonitors(detectors, monitors);
  return (
    <EuiDataGrid
      aria-label="Detector names and monitors"
      columns={[
        {
          id: 'Detector',
          isResizable: false,
          isExpandable: false,
          isSortable: false,
        },
        {
          id: 'Monitor',
          isResizable: false,
          isExpandable: false,
          isSortable: false,
        },
      ]}
      columnVisibility={{
        visibleColumns: ['Detector', 'Monitor'],
        setVisibleColumns: () => {},
      }}
      rowCount={gridData.length}
      renderCellValue={({ rowIndex, columnId }) =>
        //@ts-ignore
        gridData[rowIndex][columnId]
      }
      gridStyle={{
        border: 'horizontal',
        header: 'shade',
        rowHover: 'highlight',
        stripes: true,
      }}
      toolbarVisibility={false}
    />
  );
};

export const getNamesAndMonitorsAndStatesGrid = (
  detectors: DetectorListItem[],
  monitors: { [key: string]: Monitor }
) => {
  const gridData = getNamesAndMonitorsAndStates(detectors, monitors);
  return (
    <EuiDataGrid
      aria-label="Detector names and monitors and state"
      columns={[
        {
          id: 'Detector',
          isResizable: false,
          isExpandable: false,
          isSortable: false,
        },
        {
          id: 'Monitor',
          isResizable: false,
          isExpandable: false,
          isSortable: false,
        },
        {
          id: 'Running',
          isResizable: false,
          isExpandable: false,
          isSortable: false,
        },
      ]}
      columnVisibility={{
        visibleColumns: ['Detector', 'Monitor', 'Running'],
        setVisibleColumns: () => {},
      }}
      rowCount={gridData.length}
      renderCellValue={({ rowIndex, columnId }) =>
        //@ts-ignore
        gridData[rowIndex][columnId]
      }
      gridStyle={{
        border: 'horizontal',
        header: 'shade',
        rowHover: 'highlight',
        stripes: true,
      }}
      toolbarVisibility={false}
    />
  );
};

export const containsEnabledDetectors = (detectors: DetectorListItem[]) => {
  const enabledDetectors = detectors.filter(
    (detector) =>
      detector.curState === DETECTOR_STATE.RUNNING ||
      detector.curState === DETECTOR_STATE.INIT
  );
  return !isEmpty(enabledDetectors);
};
