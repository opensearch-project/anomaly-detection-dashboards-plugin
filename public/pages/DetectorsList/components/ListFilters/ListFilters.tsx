/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import {
  EuiComboBox,
  EuiComboBoxOptionProps,
  EuiFieldSearch,
  EuiFlexGroup,
  EuiFlexItem,
  EuiPagination,
} from '@elastic/eui';
import React from 'react';
import { getDetectorStateOptions } from '../../utils/helpers';
import { DETECTOR_STATE } from '../../../../utils/constants';

interface ListFiltersProps {
  activePage: number;
  pageCount: number;
  search: string;
  selectedDetectorStates: DETECTOR_STATE[];
  selectedIndices: string[];
  indexOptions: EuiComboBoxOptionProps[];
  onDetectorStateChange: (options: EuiComboBoxOptionProps[]) => void;
  onIndexChange: (options: EuiComboBoxOptionProps[]) => void;
  onSearchDetectorChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onSearchIndexChange: (searchValue: string) => void;
  onPageClick: (pageNumber: number) => void;
}
export const ListFilters = (props: ListFiltersProps) => (
  <EuiFlexGroup gutterSize="s">
    <EuiFlexItem grow={false} style={{ width: '40%' }}>
      <EuiFieldSearch
        fullWidth={true}
        value={props.search}
        placeholder="Search"
        onChange={props.onSearchDetectorChange}
        data-test-subj="detectorListSearch"
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <EuiComboBox
        id="selectedDetectorStates"
        data-test-subj="detectorStateFilter"
        placeholder="All detector states"
        isClearable={true}
        singleSelection={false}
        options={getDetectorStateOptions()}
        onChange={props.onDetectorStateChange}
        selectedOptions={
          props.selectedDetectorStates.length > 0
            ? props.selectedDetectorStates.map((index) => ({ label: index }))
            : []
        }
        fullWidth={true}
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <EuiComboBox
        id="selectedIndices"
        data-test-subj="indicesFilter"
        placeholder="All indices"
        isClearable={true}
        singleSelection={false}
        options={props.indexOptions}
        onChange={props.onIndexChange}
        onSearchChange={props.onSearchIndexChange}
        selectedOptions={
          props.selectedIndices.length > 0
            ? props.selectedIndices.map((index) => ({ label: index }))
            : []
        }
        fullWidth={true}
      />
    </EuiFlexItem>
    {props.pageCount > 1 ? (
      <EuiFlexItem grow={false} style={{ justifyContent: 'center' }}>
        <EuiPagination
          pageCount={props.pageCount}
          activePage={props.activePage}
          onPageClick={props.onPageClick}
          data-test-subj="detectorPageControls"
        />
      </EuiFlexItem>
    ) : null}
  </EuiFlexGroup>
);
