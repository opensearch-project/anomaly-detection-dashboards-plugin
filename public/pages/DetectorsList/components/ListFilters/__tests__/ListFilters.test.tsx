/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import { fireEvent, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { ALL_DETECTOR_STATES, ALL_INDICES } from '../../../../utils/constants';
import { ListFilters } from '../ListFilters';
import { DETECTOR_STATE } from '../../../../../../server/utils/constants';

describe('<ListFilters /> spec', () => {
  const defaultProps = {
    activePage: 1,
    pageCount: 10,
    search: '',
    selectedDetectorStates: ALL_DETECTOR_STATES,
    selectedIndices: ALL_INDICES,
    detectorStateOptions: [],
    indexOptions: [],
    onDetectorStateChange: jest.fn(),
    onIndexChange: jest.fn(),
    onSearchDetectorChange: jest.fn(),
    onSearchIndexChange: jest.fn(),
    onPageClick: jest.fn(),
  };
  beforeEach(() => {
    jest.clearAllMocks();
  });
  describe('Empty results', () => {
    test('renders component with empty message', async () => {
      const { container } = render(<ListFilters {...defaultProps} />);
      expect(container.firstChild).toMatchSnapshot();
    });
    test('should call onSearchDetectorChange callback when user inputs text', () => {
      const { getByPlaceholderText } = render(
        <ListFilters {...defaultProps} />
      );
      userEvent.type(getByPlaceholderText('Search'), 'Testing');
      expect(defaultProps.onSearchDetectorChange).toHaveBeenCalledTimes(7);
    });
    test('pagination should be hidden if pages count is 1', async () => {
      const { queryByTestId } = render(
        <ListFilters {...defaultProps} pageCount={1} />
      );
      expect(queryByTestId('detectorPageControls')).toBeNull();
    });
    test('pagination should be visible if pages count more than 1', async () => {
      const { queryByTestId } = render(
        <ListFilters {...defaultProps} pageCount={2} />
      );
      expect(queryByTestId('detectorPageControls')).not.toBeNull();
    });
    test('should call onPageClick when click on the page', async () => {
      const { getByTestId } = render(<ListFilters {...defaultProps} />);
      fireEvent.click(getByTestId('pagination-button-3'));
      expect(defaultProps.onPageClick).toHaveBeenCalledTimes(1);
    });
    test('should display default detector state and index options', () => {
      const { getByText } = render(<ListFilters {...defaultProps} />);
      expect(getByText('All detector states')).toBeInTheDocument();
      expect(getByText('All indices')).toBeInTheDocument();
    });
    test('should display selected detector state and index options', () => {
      const updatedProps = {
        ...defaultProps,
        selectedDetectorStates: [DETECTOR_STATE.DISABLED],
        selectedIndices: ['test_index'],
      };
      const { getByText } = render(<ListFilters {...updatedProps} />);
      expect(getByText(DETECTOR_STATE.DISABLED)).toBeInTheDocument();
      expect(getByText('test_index')).toBeInTheDocument();
    });
    test('should call onIndexSearchChange when searching in index filter', () => {
      const { getAllByTestId } = render(<ListFilters {...defaultProps} />);
      userEvent.type(getAllByTestId('comboBoxSearchInput')[1], 'Testing');
      expect(defaultProps.onSearchIndexChange).toHaveBeenCalledTimes(7);
    });
    test('should display multiple selected detector state and index options', () => {
      const updatedProps = {
        ...defaultProps,
        selectedDetectorStates: [DETECTOR_STATE.DISABLED, DETECTOR_STATE.INIT],
        selectedIndices: ['test_index_1', 'test_index_2'],
      };
      const { getByText } = render(<ListFilters {...updatedProps} />);
      expect(getByText(DETECTOR_STATE.DISABLED)).toBeInTheDocument();
      expect(getByText(DETECTOR_STATE.INIT)).toBeInTheDocument();
      expect(getByText('test_index_1')).toBeInTheDocument();
      expect(getByText('test_index_2')).toBeInTheDocument();
    });
  });
});
