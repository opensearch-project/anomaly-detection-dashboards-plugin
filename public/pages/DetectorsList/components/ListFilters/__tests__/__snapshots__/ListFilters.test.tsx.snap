// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListFilters /> spec Empty results renders component with empty message 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--directionRow euiFlexGroup--responsive"
>
  <div
    class="euiFlexItem euiFlexItem--flexGrowZero"
    style="width: 40%;"
  >
    <div
      class="euiFormControlLayout euiFormControlLayout--fullWidth"
    >
      <div
        class="euiFormControlLayout__childrenWrapper"
      >
        <input
          class="euiFieldSearch euiFieldSearch--fullWidth"
          data-test-subj="detectorListSearch"
          placeholder="Search"
          type="search"
          value=""
        />
        <div
          class="euiFormControlLayoutIcons"
        >
          <span
            class="euiFormControlLayoutCustomIcon"
          >
            <svg
              aria-hidden="true"
              class="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
              focusable="false"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiFlexItem"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      class="euiComboBox euiComboBox--fullWidth"
      data-test-subj="detectorStateFilter"
      role="combobox"
    >
      <div
        class="euiFormControlLayout euiFormControlLayout--fullWidth"
      >
        <div
          class="euiFormControlLayout__childrenWrapper"
        >
          <div
            class="euiComboBox__inputWrap euiComboBox__inputWrap--fullWidth euiComboBox__inputWrap-isClearable"
            data-test-subj="comboBoxInput"
            tabindex="-1"
          >
            <p
              class="euiComboBoxPlaceholder"
            >
              All detector states
            </p>
            <div
              class="euiComboBox__input"
              style="font-size: 14px; display: inline-block;"
            >
              <input
                aria-controls=""
                data-test-subj="comboBoxSearchInput"
                id="selectedDetectorStates"
                role="textbox"
                style="box-sizing: content-box; width: 2px;"
                value=""
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
              />
            </div>
          </div>
          <div
            class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
          >
            <button
              aria-label="Open list of options"
              class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
              data-test-subj="comboBoxToggleListButton"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiFlexItem"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      class="euiComboBox euiComboBox--fullWidth"
      data-test-subj="indicesFilter"
      role="combobox"
    >
      <div
        class="euiFormControlLayout euiFormControlLayout--fullWidth"
      >
        <div
          class="euiFormControlLayout__childrenWrapper"
        >
          <div
            class="euiComboBox__inputWrap euiComboBox__inputWrap--fullWidth euiComboBox__inputWrap-isClearable"
            data-test-subj="comboBoxInput"
            tabindex="-1"
          >
            <p
              class="euiComboBoxPlaceholder"
            >
              All indices
            </p>
            <div
              class="euiComboBox__input"
              style="font-size: 14px; display: inline-block;"
            >
              <input
                aria-controls=""
                data-test-subj="comboBoxSearchInput"
                id="selectedIndices"
                role="textbox"
                style="box-sizing: content-box; width: 2px;"
                value=""
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
              />
            </div>
          </div>
          <div
            class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
          >
            <button
              aria-label="Open list of options"
              class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
              data-test-subj="comboBoxToggleListButton"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiFlexItem euiFlexItem--flexGrowZero"
    style="justify-content: center;"
  >
    <nav
      class="euiPagination"
      data-test-subj="detectorPageControls"
    >
      <button
        aria-label="Previous page, 1"
        class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall"
        data-test-subj="pagination-button-previous"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonIcon__icon"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
          />
        </svg>
      </button>
      <ul
        class="euiPagination__list"
      >
        <li
          class="euiPagination__item"
        >
          <button
            aria-label="Page 1 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--small euiPaginationButton euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-0"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                1
              </span>
            </span>
          </button>
        </li>
        <li
          class="euiPagination__item"
        >
          <button
            aria-current="true"
            aria-label="Page 2 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--small euiButtonEmpty-isDisabled euiPaginationButton euiPaginationButton-isActive euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-1"
            disabled=""
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                2
              </span>
            </span>
          </button>
        </li>
        <li
          class="euiPagination__item"
        >
          <button
            aria-label="Page 3 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--small euiPaginationButton euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-2"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                3
              </span>
            </span>
          </button>
        </li>
        <li
          class="euiPagination__item"
        >
          <button
            aria-label="Page 4 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--small euiPaginationButton euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-3"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                4
              </span>
            </span>
          </button>
        </li>
        <li
          class="euiPagination__item"
        >
          <button
            aria-label="Page 5 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--small euiPaginationButton euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-4"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                5
              </span>
            </span>
          </button>
        </li>
        <li
          aria-label="Skipping pages 6 to 9"
          class="euiPaginationButton-isPlaceholder euiPagination__item"
        >
          …
        </li>
        <li
          class="euiPagination__item"
        >
          <button
            aria-label="Page 10 of 10"
            class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--small euiPaginationButton euiPaginationButton--hideOnMobile"
            data-test-subj="pagination-button-9"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                10
              </span>
            </span>
          </button>
        </li>
      </ul>
      <button
        aria-label="Next page, 3"
        class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall"
        data-test-subj="pagination-button-next"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonIcon__icon"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
          />
        </svg>
      </button>
    </nav>
  </div>
</div>
`;
